capten/agent/internal/crossplane/package_providers.go

```diff
@@ -162,3 +174,39 @@ func (h *ProvidersSyncHandler) updateCrossplaneProvider(k8sProviders []model.Pro
 	}
 	return nil
 }
+
+func (h *ProvidersSyncHandler) updateCrossplaneProviderToOutOfSync(k8sProvider *model.Provider) error {
+	dbProviders, err := h.dbStore.GetCrossplaneProviders()
+	if err != nil {
+		return fmt.Errorf("failed to get Crossplane Providers, %v", err)
+	}
+

```

Comment:
merge with existing update function, and get from DB for this specific provider.